llvm/lib/Target/RISCV/RISCVInstrInfoVPseudos.td

```diff
@@ -6417,7 +6429,7 @@ let Predicates = [HasVInstructionsAnyF] in {
 //===----------------------------------------------------------------------===//
 // 13.2. Vector Single-Width Floating-Point Add/Subtract Instructions
 //===----------------------------------------------------------------------===//
-let mayRaiseFPException = true, hasPostISelHook = 1 in {
+let mayRaiseFPException = true, hasPostISelHook = 1, hasSideEffects = 0 in {
```

Comment:
Why do we need this change?

Reply:
Without it, I was getting 
```
/home/mmaitland/repos/llvm-project/llvm/lib/Target/RISCV/RISCVInstrInfoVPseudos.td:2146:9: error: Can't infer hasSideEffects from patterns
    def suffix : VPseudoBinaryNoMaskRoundingMode<RetClass, Op1Class, Op2Class,
```
 I was surprised that these changes caused that error if it was not happening before.

Reply:
The one below appears to be unnecessary... I'll take a look at why you're getting that error.

Reply:
I suspect your issue is because you're generating unneeded pseudos like `PseudoVFADD_VFPR16_M1_E32`. FPR16 can only be used with SEW=16. So effectively it was already SEW aware.

Reply:
I am going to do some revisions to include the changes @wangpc-pp proposed in https://reviews.llvm.org/D153000. I think after I follow the approach there my problems may go away too.